### PR TITLE
Do not append highlight- on 'none' captions

### LINKED ISSUE #37164: [BUG] Azure.Search.Documents creates invalid request payload for Semantic searches
### Library name and version

Azure.Search.Documents 11.5.0-beta.2

### Describe the bug

When passing QueryCaptionsType.None, it gets serialized down to `none|highlight-false`/`none|highlight-true`, which is an invalid response to the service. `none` does not accept any highlight parameters.

Mentioned in: https://github.com/Azure-Samples/azure-search-openai-demo-csharp/issues/113

### Expected behavior

The SDK to properly serialize the value in JSON.

### Actual behavior

The SDK crafts an invalid HTTP request payload which results in a 400 BadRequest response from the service.

### Reproduction Steps

Create a semantic request with options for captioning set to None, for example:

```
new SearchOptions
            {
                Filter = filter,
                QueryType = SearchQueryType.Semantic,
                QueryLanguage = "en-us",
                QuerySpeller = "lexicon",
                SemanticConfigurationName = "default",
                Size = top,
                QueryCaption = QueryCaptionType.None,
                QueryCaptionHighlightEnabled = false
            }
```

It will fail.

### Environment

Off-topic: If you could please link the source of where the exact JSON serialization was done? I tracked the code by debugging into it, but I never found the contract responsible for the custom struct serialization.

Thanks.

### PR DESCRIPTION
**Pull request by kikaragyozov**
Fixes #37164 .

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs [hunk 1]
```diff
@@ -287,7 +287,14 @@ internal string QueryCaptionRaw
...
                 {
-                    queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
+                    if (QueryCaption.Value != QueryCaptionType.None)
+                    {
+                        queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
+                    }
+                    else
+                    {
+                        queryCaptionStringValue = QueryCaption.Value.ToString();
+                    }
```
Comment (ShivangiReja) [inline sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs:297]:
```suggestion
                    if (QueryCaption.Value == QueryCaptionType.Extractive)
                    {
                        queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
                    }
                    else
                    {
                        queryCaptionStringValue = QueryCaption.Value.ToString();
                    }
```

As per the docs [here](https://learn.microsoft.com/en-us/dotnet/api/azure.search.documents.models.querycaptiontype?view=azure-dotnet-preview#definition):

>When Captions is set to 'extractive', highlighting is enabled by default, and can be configured by appending the pipe character '|' followed by the 'highlight-<true/false>' option, such as 'extractive|highlight-true'. Defaults to 'None'

The condition should be:
```csharp
if (QueryCaption.Value == QueryCaptionType.Extractive)
{
     queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
}
else
 {
      queryCaptionStringValue = QueryCaption.Value.ToString();
 }
```

**Comment by ShivangiReja** [comment]
@kikaragyozov, we appreciate your contribution! 

As a result of your changes, a few unit test cases have started to fail. Specifically, the [QueryCaptionOption](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs#L233) and [QueryCaptionOptionWithNoHighlight](https://chat.openai.com/QueryCaptionOptionWithNoHighlight) tests. Could you please review the [SearchOptionsTests](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs) file?

**Comment by kikaragyozov** [comment]
> @kikaragyozov, we appreciate your contribution!
> 
> As a result of your changes, a few unit test cases have started to fail. Specifically, the [QueryCaptionOption](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs#L233) and [QueryCaptionOptionWithNoHighlight](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs#L193) tests. Could you please review the [SearchOptionsTests](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs) file?

Sorry about the inverted if logic! My brain must have been asleep at the time.

**Comment by ShivangiReja** [comment]
Hello @kikaragyozov, we have plans to include this bug fix in our upcoming preview release scheduled for next week. If, for any reason, you are unable to address the unit test case modifications as mentioned earlier, I can incorporate this bug fix along with the necessary unit test case updates in my upcoming pull request. Thank you!

**Comment by kikaragyozov** [comment]
> Thank you for taking the time to update the code!
> 
> > As a result of your changes, a few unit test cases have started to fail. Specifically, the [QueryCaptionOption](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs#L233) and [QueryCaptionOptionWithNoHighlight](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs#L193) tests. Could you please review the [SearchOptionsTests](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs) file?
> 
> @kikaragyozov, in order to pass all the checks, we need to update the unit test cases as mentioned above. Please let me know if you require any assistance. Thank you!

I had a busy couple of days. I'll get on it now! Thank you for the patience. If I have any questions I'll be sure to ask.

File: sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs [hunk 2]
```diff
@@ -253,10 +253,10 @@ public void QueryCaptionOption()
 
             // We can also set `QueryCaption` to a value unknown to the SDK.
             searchOptions.QueryCaption = "unknown";
-            Assert.AreEqual($"unknown|highlight-True", searchOptions.QueryCaptionRaw);
```
Comment (kikaragyozov) [inline sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs:256]: Do we want this to happen? Or should we modify the code to allow for passing of `highlight` parameter when `unknown` is passed? I couldn't test whether the SDK is fine with accepting an `unknown|highlight-True` payload.

@ShivangiReja thoughts?

**Comment by kikaragyozov** [comment]
> @kikaragyozov please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement



> @kikaragyozov please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs (hunk 2 above)
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs:256]: No, we want to allow the `highlight` parameter to be passed only when the `QueryCaptionType` is set to `Extractive`. The code is correct in this PR.

**Review by ShivangiReja — APPROVED** [review]
Thank you @kikaragyozov for your contribution!

**Comment by kikaragyozov** [comment]
> Thank you @kikaragyozov for your contribution!

Thank you for the patience! :)

**Pull request merged**